ocpp-transport-soap/src/test/kotlin/com/izivia/ocpp/http/test/HttpClientTest.kt

```diff
@@ -0,0 +1,193 @@
+                        <MessageID xmlns="http://www.w3.org/2005/08/addressing">uuid:b70e2eae-bd09-4546-b049-8b2be7ccf578</MessageID>
+                    </S:Header>
+                    <S:Body>
+                        <heartbeatRequest xmlns="urn://Ocpp/Cs/2015/10/"/>
+                    </S:Body>
+                </S:Envelope>
+            """.trimIndent()
+            )
+
+        val result = JavaHttpClient().invoke(request)
+
+        expectThat(result) {
+            get { bodyString() }.isNotBlank()
+            get { status }.isEqualTo(Status.OK)
+        }
+        expectThat(received).isTrue()
+
+        client.close()
+    }
+
+    @Test
+    fun `should not receive the OCPP 1-6 HeartBeat request because there is no handler`() {
+        val received = false
+        val client = HttpClient(
+            path = "/ocpp/soap",
+            port = 5001,
+            ocppId = "ocppChargePoint",
+            target = "http://localhost:5002/api",
+            ocppSoapParser = Ocpp16SoapParser()
+        )
+        client.connect()
+
+        val request = Request(Method.POST, "http://127.0.0.1:5001/ocpp/soap/HeartBeat/1234")
+            .body(
+                """
+                <S:Envelope xmlns:S="http://www.w3.org/2003/05/soap-envelope">
+                    <S:Header>
+                        <chargeBoxIdentity xmlns="urn://Ocpp/Cs/2015/10/">92073_05_04</chargeBoxIdentity>
+                        <wsa:From xmlns:wsa="http://www.w3.org/2005/08/addressing">
+                            <wsa:Address>http://192.168.0.3:12800/ws</wsa:Address>
+                        </wsa:From>
+                        <To xmlns="http://www.w3.org/2005/08/addressing">http://sigeif-enovates.vpn.l30.sodetrel.fr/ocpp/v16s/</To>
+                        <Action xmlns="http://www.w3.org/2005/08/addressing">/Heartbeat</Action>
+                        <ReplyTo xmlns="http://www.w3.org/2005/08/addressing">
+                            <Address>http://www.w3.org/2005/08/addressing/anonymous</Address>
+                        </ReplyTo>
+                        <FaultTo xmlns="http://www.w3.org/2005/08/addressing">
+                            <Address>http://www.w3.org/2005/08/addressing/anonymous</Address>
+                        </FaultTo>
+                        <MessageID xmlns="http://www.w3.org/2005/08/addressing">uuid:b70e2eae-bd09-4546-b049-8b2be7ccf578</MessageID>
+                    </S:Header>
+                    <S:Body>
+                        <heartbeatRequest xmlns="urn://Ocpp/Cs/2015/10/"/>
+                    </S:Body>
+                </S:Envelope>
+            """.trimIndent()
+            )
+
+        val result = JavaHttpClient().invoke(request)
+
+        expectThat(result) {
+            get { bodyString() }.isBlank()
+            get { status }.isEqualTo(Status.NOT_FOUND)
+        }
+        expectThat(received).isFalse()
+
+        client.close()
+    }
+
+    private fun createServer(handlerBody: String) =
```

Comment:
I would rename it to `createMockServerAndAlwaysRespond` to make it more explicit what it does. At first glance in the test I thought we were creating an occp server